src/transpilation/resolve.ts

```diff
@@ -0,0 +1,206 @@
+                const dependencyContent = emitHost.readFile(resolvedDependency);
+                if (dependencyContent === undefined) {
+                    diagnostics.push(couldNotReadDependency(resolvedDependency));
+                    continue;
+                }
+
+                const dependency = {
+                    fileName: resolvedDependency,
+                    code: dependencyContent,
+                };
+                const nestedDependencies = resolveFileDependencies(dependency, program, emitHost);
+                dependencies.push(dependency, ...nestedDependencies.resolvedFiles);
+                diagnostics.push(...nestedDependencies.diagnostics);
+            }
+        } else {
+            // Could not resolve dependency, add a diagnostic and make some fallback path
+            diagnostics.push(couldNotResolveRequire(required, path.relative(projectRootDir, file.fileName)));
+
+            const fallbackRequire = fallbackResolve(required, projectRootDir, fileDir);
+            replaceRequireInCode(file, required, fallbackRequire);
+            replaceRequireInSourceMap(file, required, fallbackRequire);
+        }
+    }
+    return { resolvedFiles: dependencies, diagnostics };
+}
+
+function replaceRequireInCode(file: ProcessedFile, originalRequire: string, newRequire: string): void {
+    const requirePath = formatPathToLuaPath(newRequire.replace(".lua", ""));
+    file.code = file.code.replace(`require("${originalRequire}")`, `require("${requirePath}")`);
+}
+
+function replaceRequireInSourceMap(file: ProcessedFile, originalRequire: string, newRequire: string): void {
+    const requirePath = formatPathToLuaPath(newRequire.replace(".lua", ""));
+    if (file.sourceMapNode) {
+        replaceInSourceMap(file.sourceMapNode, file.sourceMapNode, `"${originalRequire}"`, `"${requirePath}"`);
+    }
+}
+
+function replaceInSourceMap(node: SourceNode, parent: SourceNode, require: string, resolvedRequire: string): boolean {
+    if ((!node.children || node.children.length === 0) && node.toString() === require) {
+        parent.children = [new SourceNode(node.line, node.column, node.source, [resolvedRequire])];
+        return true; // Stop after finding the first occurrence
+    }
+
+    if (node.children) {
+        for (const c of node.children) {
+            if (replaceInSourceMap(c, node, require, resolvedRequire)) {
+                return true; // Occurrence found in one of the children
+            }
+        }
+    }
+
+    return false; // Did not find the require
+}
+
+function findRequiredPaths(code: string): string[] {
+    // Find all require("<path>") paths in the code
+    const paths: string[] = [];
+    const pattern = /require\("(.+)"\)/g;
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    let match: RegExpExecArray | null;
+    while ((match = pattern.exec(code))) {
+        paths.push(match[1]);
+    }
+
+    return paths;
+}
+
+function resolveDependency(
```

Comment:
The only place where this function is used is inside resolveFileDependencies (https://github.com/TypeScriptToLua/TypeScriptToLua/pull/1011/files#diff-d2c750f93b463ac383b3f4a3127e379381ce0e0ff1a5934c12bd2216f84d4125R53) Therefore I think this function should be moved closer/below resolveFileDependencies, to improve readability.